src/Functions/if.cpp

```diff
@@ -45,52 +46,86 @@ using namespace GatherUtils;
   * then, else - numeric types for which there is a general type, or dates, datetimes, or strings, or arrays of these types.
   */
 
+template <typename ArrayCond, typename ArrayA, typename ArrayB, typename ArrayResult, typename ResultType>
+static inline void fillVectorVector(const ArrayCond & cond, const ArrayA & a, const ArrayB & b, ArrayResult & res)
+{
+    size_t size = cond.size();
+    bool a_is_short = a.size() < size;
+    bool b_is_short = b.size() < size;
+
+    size_t a_index = 0;
+    size_t b_index = 0;
+
+    for (size_t i = 0; i < size; ++i)
+    {
+        if (cond[i])
+            res[i] = a_is_short ? static_cast<ResultType>(a[a_index++]) : static_cast<ResultType>(a[i]);
+        else
+            res[i] = b_is_short ? static_cast<ResultType>(b[b_index++]) : static_cast<ResultType>(b[i]);
+    }
```

Comment:
This I think I would implement as
```cpp
if (a_is_short && b_is_short)
{
    for (size_t i = 0; i < size; ++i)
    {
        if (cond[i])
            res[i] = static_cast<ResultType>(a[a_index++]);
        else
            res[i] = static_cast<ResultType>(b[b_index++]);
    }
}
else if (a_is_short)
{
    for (size_t i = 0; i < size; ++i)
    {
        if (cond[i])
            res[i] = static_cast<ResultType>(a[a_index++]);
        else
            res[i] = static_cast<ResultType>(b[i]);
    }
}
else if (a_is_short)
{
 ...
}
else
{
 ...
}
```
It's more code, but less conditions in a loop.
The same with `fillVectorConstant` and `fillConstantVector`